### PR TITLE
Create Event Card

### PR DESCRIPTION
**Pull request by yanbhoe**
image

### CODE REVIEW

File: web/src/data/events.js [hunk 1]
```diff
@@ -17,11 +17,21 @@ const EventTypes = Object.freeze({
...
-  constructor(type, title, date, eventSchedule, externalLinks) {
+  constructor(
+    type,
+    title,
+    date,
+    eventSchedule = [],
+    location = "",
+    price = "",
+    externalLinks = undefined
+  ) {
```
Comment (yanbhoe) [inline web/src/data/events.js:28]: This shit is why we need typescript

Comment (jesseez) [reply web/src/data/events.js:26]: fair, but this is probably the only thing that will have to look like this. Nothing else is going to be rendered dynamically

File: web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        <Box>
+          <Typography sx={eventCardStyles.priceLabel}>Price:</Typography>
+          <Typography sx={eventCardStyles.price}>{props.price}</Typography>
+        </Box>
+      )}
+      {props.link && (
+        <Button
+          sx={eventCardStyles.button}
+          variant="contained"
+          onClick={() => window.open(props.link)}
```
Comment (jesseez) [inline web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.js:37]: Why not href?

File: web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        >
+          {"View Details"}
+        </Button>
+      )}
+    </Box>
+  </Box>
+);
+
+const EventCard = (props) => {
+  let link;
```
Comment (jesseez) [inline web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.js:47]: Can we hardcode this to just be the FB link for now? We can update the code once we support other link types

File: web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.styles.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    fontWeight: "600",
+    color: theme.palette.text.primary,
+  },
+  footerBox: {
+    display: "flex",
+    justifyContent: "space-between",
+  },
+  button: {
+    backgroundImage: "linear-gradient(#8637DA, #6400CF)",
+    width: "135px",
```
Comment (jesseez) [inline web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.styles.js:45]: intervals of 8?

File: web/src/data/events.js [hunk 5]
```diff
@@ -31,6 +41,11 @@ const defaultDyosSchedule = [
   { time: "9:00pm - 11:30pm", name: "Social Dancing" },
 ];
 
+const addons = [
```
Comment (jesseez) [inline web/src/data/events.js:44]: What is this? Looks like it's not used anywhere

File: web/src/data/events.js [hunk 6]
```diff
@@ -41,13 +56,17 @@ const events = [
     EventTypes.DYOS,
     messages.doYourOwnSwing,
     createDate("05/16/2024"),
-    defaultDyosSchedule
+    defaultDyosSchedule,
+    "Studio M Ballroom, San Jose, CA",
```
Comment (jesseez) [inline web/src/data/events.js:60]: Can we put this in common/messages.js?

File: web/src/data/events.js [hunk 7]
```diff
@@ -65,31 +86,39 @@ const events = [
...
   new Event(
     EventTypes.DYOS,
     "DYOS: Slingshot from Any Hand",
     createDate("06/13/2024"),
     defaultDyosSchedule,
+    "Studio M Ballroom, San Jose, CA",
+    "$10 - $40",
     {
-      facebookLink: "https://www.facebook.com/events/976640170426321",
+      facebookLink: "https://www.facebook.com/events/457255416887420",
```
Comment (jesseez) [inline web/src/data/events.js:104]: Undo this change in link

File: web/src/data/events.js [hunk 8]
```diff
@@ -98,7 +127,12 @@ const events = [
     EventTypes.DYOS,
     messages.doYourOwnSwing,
     createDate("07/11/2024"),
-    defaultDyosSchedule
+    defaultDyosSchedule,
+    "Studio M Ballroom, San Jose, CA",
+    "$10 - $40",
+    {
+      facebookLink: "https://www.facebook.com/events/457255416887420",
```
Comment (jesseez) [inline web/src/data/events.js:134]: Can we remove this so we know we need to update it?

**Review by jesseez — CHANGES_REQUESTED** [review]
Overall LGTM. Thank you for your service

File: web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.js (hunk 2 above)
Comment (yanbhoe) [reply web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.js:37]: doesn't work

File: web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.styles.js (hunk 4 above)
Comment (yanbhoe) [reply web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.styles.js:45]: I did these pixels based on vedant's figma, they were not intervals of 8 but i can make them that way

File: web/src/data/events.js (hunk 5 above)
Comment (yanbhoe) [reply web/src/data/events.js:44]: Oh I thought I removed this

File: web/src/data/events.js (hunk 7 above)
Comment (yanbhoe) [reply web/src/data/events.js:104]: oh mb

File: web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.js (hunk 2 above)
Comment (yanbhoe) [reply web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.js:37]: nvm it does

File: web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.styles.js (hunk 4 above)
Comment (yanbhoe) [reply web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.styles.js:45]: but 136 is probably not gonna drastically change it

File: web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.js [hunk 9]
```diff
@@ -34,7 +34,9 @@ const Footer = (props) => (
         <Button
           sx={eventCardStyles.button}
           variant="contained"
-          onClick={() => window.open(props.link)}
+          // onClick={() => window.open(props.link)}
```
Comment (jesseez) [inline web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.js:37]: Remove - also want to set `rel="noreferrer noopener"`

File: web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.styles.js [hunk 10]
```diff
@@ -42,7 +42,7 @@ const eventCardStyles = {
   },
   button: {
     backgroundImage: "linear-gradient(#8637DA, #6400CF)",
-    width: "135px",
+    width: "136px",
     height: "51px",
```
Comment (jesseez) [inline web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.styles.js:46]: this one too please

**Review by jesseez — APPROVED** [review]
Ty

File: web/src/data/events.js (hunk 7 above)
Comment (yanbhoe) [inline web/src/data/events.js]:
```suggestion
      facebookLink: "https://www.facebook.com/events/976640170426321",
```

File: web/src/data/events.js [hunk 11]
```diff
@@ -65,31 +86,39 @@ const events = [
...
   new Event(
     EventTypes.DYOS,
     "DYOS: Wraps and Rolls Galore!",
     createDate("06/20/2024"),
     defaultDyosSchedule,
-    {
-      facebookLink: "https://www.facebook.com/events/476321498262098",
-    }
+    "Studio M Ballroom, San Jose, CA",
+    "$10 - $40"
```
Comment (yanbhoe) [inline web/src/data/events.js]:
```suggestion
    "$10 - $40",
    {
      facebookLink: "https://www.facebook.com/events/476321498262098",
    }
```

File: web/src/data/events.js [hunk 12]
```diff
@@ -81,15 +108,16 @@ const events = [
     "DYOS: Wraps and Rolls Galore!",
     createDate("06/20/2024"),
     defaultDyosSchedule,
-    {
-      facebookLink: "https://www.facebook.com/events/476321498262098",
-    }
+    DEFAULT_LOCATION,
+    DEFAULT_PRICE
```
Comment (yanbhoe) [inline web/src/data/events.js:112]:
```suggestion
    DEFAULT_PRICE,
    {
      facebookLink: "https://www.facebook.com/events/476321498262098",
    }
```

File: web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.js [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+        <Box>
+          <Typography sx={eventCardStyles.priceLabel}>Price:</Typography>
+          <Typography sx={eventCardStyles.price}>{props.price}</Typography>
+        </Box>
+      )}
+      {props.link && (
+        <Button
+          sx={eventCardStyles.button}
+          variant="contained"
+          // onClick={() => window.open(props.link)}
```
Comment (yanbhoe) [inline web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.js:37]:
```suggestion
          rel="noreferrer noopener"
```

File: web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.styles.js [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+    color: theme.palette.text.primary,
+  },
+  footerBox: {
+    display: "flex",
+    justifyContent: "space-between",
+  },
+  button: {
+    backgroundImage: "linear-gradient(#8637DA, #6400CF)",
+    width: "136px",
+    height: "51px",
```
Comment (yanbhoe) [inline web/src/components/home/subcomponents/upcoming_events/event_cards/event_card.styles.js:46]:
```suggestion
    height: "48px",
```

File: web/src/data/events.js [hunk 15]
```diff
@@ -65,13 +85,20 @@ const events = [
     EventTypes.DYOS,
     messages.doYourOwnSwing,
     createDate("06/06/2024"),
-    defaultDyosSchedule
+    defaultDyosSchedule,
+    DEFAULT_LOCATION,
+    DEFAULT_PRICE,
+    {
+      facebookLink: "https://www.facebook.com/events/457255416887420",
+    }
```
Comment (yanbhoe) [inline web/src/data/events.js:93]:
```suggestion
    DEFAULT_PRICE
```

**Comment by yanbhoe** [comment]
Since these cards are smaller, do we want to make it 3 events?

**Review by Kuboosh21 — APPROVED** [review]
